### PR TITLE
Allow easy toggle between UIs and fix Entra flow in new ui

### PR DESCRIPTION
**Pull request by mclmax**
## Proposed changes

Makes it easy to toggle between the new / old UI if we need to rollback.

This also makes sure the entra token endpoint works for both UIs.

## Types of changes

What types of changes does your code introduce to this repository?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes (if applicable)
- [ ] I have run pre-commit (`pre-commit` in the repo)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Linked to the relevant github issue or github discussion

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: canarytokens/azure_css.py [hunk 1]
```diff
@@ -10,6 +10,22 @@
...
 BearerToken = str
 
+EntraTokenErrorAccessDenied = "access_denied"
+
+ENTRA_BASE_REDIRECT_URL = "/nest/entra/{status}"
+
+ENTRA_STATUS_HAS_CUSTOM_CSS_ALREADY = "has_custom_css_already"
+ENTRA_STATUS_ERROR = "error"
+ENTRA_STATUS_SUCCESS = "success"
+ENTRA_STATUS_NO_ADMIN_CONSENT = "no_admin_consent"
```
Comment (wleightond) [inline canarytokens/azure_css.py:20]: Can we use an enum for these?

Something like:
```suggestion
class EntraStatus(enum.Enum):
    HAS_CUSTOM_CSS_ALREADY = "has_custom_css_already"
    ERROR = "error"
    SUCCESS = "success"
    NO_ADMIN_CONSENT = "no_admin_consent"
    ERROR_ACCESS_DENIED = "access_denied"

ENTRA_BASE_REDIRECT_URL = "/nest/entra/{status}"
```

Then we just import that one thing on the other side and check `EntraStatus.<whatever>` instead

File: frontend/app.py [hunk 2]
```diff
@@ -575,39 +592,43 @@ async def download(
...
+# remove up until here once new ui stable
+
+
+# NOTE: Do not remove this when cleaning up after UI is stable
 @app.get("/azure_css_landing", tags=["Azure Portal Phishing Protection App"])
 async def azure_css_landing(
-    request: Request, admin_consent: str = "", tenant: str = None, state: str = None
+    request: Request,
+    admin_consent: str = "",
+    tenant: str = None,
```
Comment (wleightond) [inline frontend/app.py:609]: If the default is `None` then the types here should be `Optional[str]`.

**Review by wleightond — CHANGES_REQUESTED** [review]

File: frontend/app.py [hunk 3]
```diff
@@ -575,39 +592,43 @@ async def download(
...
-    return templates.TemplateResponse(
-        "azure_install.html",
-        {"request": request, "status": info},
-    )
+        status = install_azure_css(tenant, css)
 
+    if not frontend_settings.NEW_UI:
+        return templates.TemplateResponse(
+            "azure_install.html",
+            {"request": request, "status": LEGACY_ENTRA_STATUS_MAP[status]},
```
Comment (thinkst-az) [inline frontend/app.py:628]: Consider returning for example:

```
"status":  "has_custom_css_already'
"status_msg": "Installation failed: your .... "
```

Then the frontend (or other system using the API) has a machine-checkable status if it needs to make a decision based on it, and message if it has to render one for user.

File: canarytokens/azure_css.py (hunk 1 above)
Comment (mclmax) [reply canarytokens/azure_css.py:20]: the python version we use doesn't support StrEnum, that was the plan originally. enum.StrEnum. so without that we'd have to do `EntraStatus.<whatever>.value` or:

```
class EntraStatus(str, enum.Enum):
     HAS_CUSTOM_CSS_ALREADY = "has_custom_css_already"
     ERROR = "error"
     SUCCESS = "success"
     NO_ADMIN_CONSENT = "no_admin_consent"
     ERROR_ACCESS_DENIED = "access_denied"

     def __str__(self):
         return str(self.value)```

**Review by wleightond — APPROVED** [review]

**Pull request merged**